com.unity.render-pipelines.universal/Runtime/2D/Passes/Render2DLightingPass.cs

```diff
@@ -107,55 +130,54 @@ public override void Execute(ScriptableRenderContext context, ref RenderingData
                     combinedDrawSettings.sortingSettings = sortSettings;
                     normalsDrawSettings.sortingSettings = sortSettings;
 
-                    for (int i = 0; i < m_BlendStyleInitialized.Length; ++i)
-                        m_BlendStyleInitialized[i] = false;
-                    
-                    for (int i = 0; i < s_SortingLayers.Length; i++)
+                    var blendStylesCount = m_Renderer2DData.lightBlendStyles.Length;
+                    var hasBeenInitialized = new bool[blendStylesCount];
```

Comment:
This needs to be moved out of the Execute() function, or GC alloc test will fail.

Reply:
fixed